src/TriangleNodesProvisioner.cpp

```diff
@@ -105,6 +105,61 @@ namespace blitzdg {
         }
     }
 
+    void TriangleNodesProvisioner::computeGradVandermondeMatrix(index_type N,  const real_vector_type & r, const real_vector_type & s, real_matrix_type & V2Dr, real_matrix_type & V2Ds) const {
+        const index_type Nr = r.length(0);
+        const index_type Np = (N+1)*(N+2)/2;
+
+        real_vector_type a(Nr), b(Nr);
+
+        rsToab(r, s, a, b);
+
+        index_type count = 0;
+        for (index_type i=0; i < N+1; ++i) {
+            for (index_type j=0; j < N-i+1; ++j) {
```

Comment:
You could consider writing the loop termination tests as i <= N and j <= N - i, to avoid to the plus 1, but it's not really a big deal.